### ISSUE TITLE
ERROR: Cannot install -r requirements.txt (line 18), -r requirements.txt (line 2) and numpy<1.20.0 and >=1.19.3 because these package versions have conflicting dependencies.

### ISSUE THREAD

**Issue by chunhualiao** [facebookresearch/CompilerGym#771]
## To Reproduce
I am sharing a Collab with all details:
 https://colab.research.google.com/drive/1jv981UXSBLi5LnEyG_zGUeFsb7FXQBTA?usp=sharing 

- pip install -U compiler_gym
- sudo apt install -y clang clang-format golang libjpeg-dev libtinfo5 m4 make patch zlib1g-dev tar bzip2 wget
- git clone https://github.com/facebookresearch/CompilerGym.git
- cd CompilerGym/examples
- make install

ERROR: Cannot install -r requirements.txt (line 18), -r requirements.txt (line 2) and numpy<1.20.0 and >=1.19.3 because these package versions have conflicting dependencies.

## Expected behavior

It should go through

## Environment

Please fill in this checklist:
```
CompilerGym: 0.2.4
Is debug build: N/A

Python version: 3.7.14 (default, Sep  8 2022, 00:06:44)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.10.133+-x86_64-with-Ubuntu-18.04-bionic

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.22.6
Libc version: glibc-2.26

Versions of all installed libraries:
    [pip3] absl-py==1.2.0
    [pip3] aeppl==0.0.33
    [pip3] aesara==2.7.9
    [pip3] aiohttp==3.8.3
    [pip3] aiosignal==1.2.0
    [pip3] alabaster==0.7.12
    [pip3] albumentations==1.2.1
    [pip3] altair==4.2.0
    [pip3] appdirs==1.4.4
    [pip3] arviz==0.12.1
    [pip3] astor==0.8.1
    [pip3] astropy==4.3.1
    [pip3] astunparse==1.6.3
    [pip3] async-timeout==4.0.2
    [pip3] asynctest==0.13.0
    [pip3] atari-py==0.2.9
    [pip3] atomicwrites==1.4.1
    [pip3] attrs==22.1.0
    [pip3] audioread==3.0.0
    [pip3] autograd==1.5
    [pip3] Babel==2.10.3
    [pip3] backcall==0.2.0
    [pip3] beautifulsoup4==4.6.3
    [pip3] bleach==5.0.1
    [pip3] blis==0.7.8
    [pip3] bokeh==2.3.3
    [pip3] branca==0.5.0
    [pip3] bs4==0.0.1
    [pip3] CacheControl==0.12.11
    [pip3] cached-property==1.5.2
    [pip3] cachetools==4.2.4
    [pip3] catalogue==2.0.8
    [pip3] certifi==2022.9.24
    [pip3] cffi==1.15.1
    [pip3] cftime==1.6.2
    [pip3] chardet==3.0.4
    [pip3] charset-normalizer==2.1.1
    [pip3] click==7.1.2
    [pip3] clikit==0.6.2
    [pip3] cloudpickle==1.5.0
    [pip3] cmake==3.22.6
    [pip3] cmdstanpy==1.0.7
    [pip3] colorcet==3.0.1
    [pip3] colorlover==0.3.0
    [pip3] community==1.0.0b1
    [pip3] compiler-gym==0.2.4
    [pip3] confection==0.0.2
    [pip3] cons==0.4.5
    [pip3] contextlib2==0.5.5
    [pip3] convertdate==2.4.0
    [pip3] crashtest==0.3.1
    [pip3] crcmod==1.7
    [pip3] cufflinks==0.17.3
    [pip3] cvxopt==1.3.0
    [pip3] cvxpy==1.2.1
    [pip3] cycler==0.11.0
    [pip3] cymem==2.0.6
    [pip3] Cython==0.29.32
    [pip3] daft==0.0.4
    [pip3] dask==2022.2.0
    [pip3] datascience==0.17.5
    [pip3] debugpy==1.0.0
    [pip3] decorator==4.4.2
    [pip3] defusedxml==0.7.1
    [pip3] Deprecated==1.2.13
    [pip3] descartes==1.1.0
    [pip3] dill==0.3.5.1
    [pip3] distributed==2022.2.0
    [pip3] dlib==19.24.0
    [pip3] dm-tree==0.1.7
    [pip3] docker==6.0.0
    [pip3] docutils==0.17.1
    [pip3] dopamine-rl==1.0.5
    [pip3] earthengine-api==0.1.326
    [pip3] easydict==1.10
    [pip3] ecos==2.0.10
    [pip3] editdistance==0.5.3
    [pip3] en-core-web-sm==3.4.0
    [pip3] entrypoints==0.4
    [pip3] ephem==4.1.3
    [pip3] et-xmlfile==1.1.0
    [pip3] etils==0.8.0
    [pip3] etuples==0.3.8
    [pip3] fa2==0.3.5
    [pip3] fastai==2.7.9
    [pip3] fastcore==1.5.27
    [pip3] fastdownload==0.0.7
    [pip3] fastdtw==0.3.4
    [pip3] fasteners==0.18
    [pip3] fastjsonschema==2.16.2
    [pip3] fastprogress==1.0.3
    [pip3] fastrlock==0.8
    [pip3] feather-format==0.4.1
    [pip3] filelock==3.8.0
    [pip3] firebase-admin==4.4.0
    [pip3] fix-yahoo-finance==0.0.22
    [pip3] Flask==1.1.4
    [pip3] flatbuffers==22.9.24
    [pip3] folium==0.12.1.post1
    [pip3] frozenlist==1.3.1
    [pip3] fsspec==2022.8.2
    [pip3] future==0.16.0
    [pip3] gast==0.5.3
    [pip3] GDAL==2.2.2
    [pip3] gdown==4.4.0
    [pip3] gensim==3.6.0
    [pip3] geographiclib==1.52
    [pip3] geopy==1.17.0
    [pip3] gin-config==0.5.0
    [pip3] glob2==0.7
    [pip3] google==2.0.3
    [pip3] google-api-core==1.31.6
    [pip3] google-api-python-client==1.12.11
    [pip3] google-auth==1.35.0
    [pip3] google-auth-httplib2==0.0.4
    [pip3] google-auth-oauthlib==0.4.6
    [pip3] google-cloud-bigquery==1.21.0
    [pip3] google-cloud-bigquery-storage==1.1.2
    [pip3] google-cloud-core==1.0.3
    [pip3] google-cloud-datastore==1.8.0
    [pip3] google-cloud-firestore==1.7.0
    [pip3] google-cloud-language==1.2.0
    [pip3] google-cloud-storage==1.18.1
    [pip3] google-cloud-translate==1.5.0
    [pip3] google-colab==1.0.0
    [pip3] google-pasta==0.2.0
    [pip3] google-resumable-media==0.4.1
    [pip3] googleapis-common-protos==1.56.4
    [pip3] googledrivedownloader==0.4
    [pip3] graphviz==0.10.1
    [pip3] greenlet==1.1.3
    [pip3] grpcio==1.43.0
    [pip3] gspread==3.4.2
    [pip3] gspread-dataframe==3.0.8
    [pip3] gym==0.20.0
    [pip3] gym-notices==0.0.8
    [pip3] h5py==3.1.0
    [pip3] HeapDict==1.0.1
    [pip3] hijri-converter==2.2.4
    [pip3] holidays==0.16
    [pip3] holoviews==1.14.9
    [pip3] html5lib==1.0.1
    [pip3] httpimport==0.5.18
    [pip3] httplib2==0.17.4
    [pip3] httplib2shim==0.0.3
    [pip3] httpstan==4.6.1
    [pip3] humanize==4.4.0
    [pip3] hyperopt==0.1.2
    [pip3] idna==2.10
    [pip3] imageio==2.9.0
    [pip3] imagesize==1.4.1
    [pip3] imbalanced-learn==0.8.1
    [pip3] imblearn==0.0
    [pip3] imgaug==0.4.0
    [pip3] importlib-metadata==5.0.0
    [pip3] importlib-resources==5.9.0
    [pip3] imutils==0.5.4
    [pip3] inflect==2.1.0
    [pip3] intel-openmp==2022.2.0
    [pip3] intervaltree==2.1.0
    [pip3] ipykernel==5.3.4
    [pip3] ipython==7.9.0
    [pip3] ipython-genutils==0.2.0
    [pip3] ipython-sql==0.3.9
    [pip3] ipywidgets==7.7.1
    [pip3] itsdangerous==1.1.0
    [pip3] jax==0.3.21
    [pip3] jaxlib==0.3.20+cuda11.cudnn805
    [pip3] jieba==0.42.1
    [pip3] Jinja2==2.11.3
    [pip3] joblib==1.2.0
    [pip3] jpeg4py==0.1.4
    [pip3] jsonschema==4.3.3
    [pip3] jupyter-client==6.1.12
    [pip3] jupyter-console==6.1.0
    [pip3] jupyter-core==4.11.1
    [pip3] jupyterlab-widgets==3.0.3
    [pip3] kaggle==1.5.12
    [pip3] kapre==0.3.7
    [pip3] keras==2.8.0
    [pip3] Keras-Preprocessing==1.1.2
    [pip3] keras-vis==0.4.1
    [pip3] kiwisolver==1.4.4
    [pip3] korean-lunar-calendar==0.3.1
    [pip3] langcodes==3.3.0
    [pip3] libclang==14.0.6
    [pip3] librosa==0.8.1
    [pip3] lightgbm==2.2.3
    [pip3] llvmlite==0.39.1
    [pip3] lmdb==0.99
    [pip3] locket==1.0.0
    [pip3] logical-unification==0.4.5
    [pip3] loop-tool-py==0.0.7
    [pip3] LunarCalendar==0.0.9
    [pip3] lxml==4.9.1
    [pip3] Markdown==3.4.1
    [pip3] MarkupSafe==2.0.1
    [pip3] marshmallow==3.18.0
    [pip3] matplotlib==3.2.2
    [pip3] matplotlib-venn==0.11.7
    [pip3] miniKanren==1.0.3
    [pip3] missingno==0.5.1
    [pip3] mistune==0.8.4
    [pip3] mizani==0.7.3
    [pip3] mkl==2019.0
    [pip3] mlxtend==0.14.0
    [pip3] more-itertools==8.14.0
    [pip3] moviepy==0.2.3.5
    [pip3] mpmath==1.2.1
    [pip3] msgpack==1.0.4
    [pip3] multidict==6.0.2
    [pip3] multipledispatch==0.6.0
    [pip3] multitasking==0.0.11
    [pip3] murmurhash==1.0.8
    [pip3] music21==5.5.0
    [pip3] natsort==5.5.0
    [pip3] nbconvert==5.6.1
    [pip3] nbformat==5.6.1
    [pip3] netCDF4==1.6.1
    [pip3] networkx==2.6.3
    [pip3] nibabel==3.0.2
    [pip3] nltk==3.7
    [pip3] notebook==5.3.1
    [pip3] numba==0.56.2
    [pip3] numexpr==2.8.3
    [pip3] numpy==1.21.6
    [pip3] oauth2client==4.1.3
    [pip3] oauthlib==3.2.1
    [pip3] okgrade==0.4.3
    [pip3] opencv-contrib-python==4.6.0.66
    [pip3] opencv-python==4.6.0.66
    [pip3] opencv-python-headless==4.6.0.66
    [pip3] openpyxl==3.0.10
    [pip3] opt-einsum==3.3.0
    [pip3] osqp==0.6.2.post0
    [pip3] packaging==21.3
    [pip3] palettable==3.3.0
    [pip3] pandas==1.3.5
    [pip3] pandas-datareader==0.9.0
    [pip3] pandas-gbq==0.13.3
    [pip3] pandas-profiling==1.4.1
    [pip3] pandocfilters==1.5.0
    [pip3] panel==0.12.1
    [pip3] param==1.12.2
    [pip3] parso==0.8.3
    [pip3] partd==1.3.0
    [pip3] pastel==0.2.1
    [pip3] pathlib==1.0.1
    [pip3] pathy==0.6.2
    [pip3] patsy==0.5.2
    [pip3] pep517==0.13.0
    [pip3] pexpect==4.8.0
    [pip3] pickleshare==0.7.5
    [pip3] Pillow==7.1.2
    [pip3] pip==21.1.3
    [pip3] pip-tools==6.2.0
    [pip3] plotly==5.5.0
    [pip3] plotnine==0.8.0
    [pip3] pluggy==0.7.1
    [pip3] pooch==1.6.0
    [pip3] portpicker==1.3.9
    [pip3] prefetch-generator==1.0.1
    [pip3] preshed==3.0.7
    [pip3] prettytable==3.4.1
    [pip3] progressbar2==3.38.0
    [pip3] promise==2.3
    [pip3] prompt-toolkit==2.0.10
    [pip3] prophet==1.1.1
    [pip3] protobuf==3.17.3
    [pip3] psutil==5.4.8
    [pip3] psycopg2==2.9.3
    [pip3] ptyprocess==0.7.0
    [pip3] py==1.11.0
    [pip3] pyarrow==6.0.1
    [pip3] pyasn1==0.4.8
    [pip3] pyasn1-modules==0.2.8
    [pip3] pybind11==2.10.0
    [pip3] pycocotools==2.0.5
    [pip3] pycparser==2.21
    [pip3] pyct==0.4.8
    [pip3] pydantic==1.9.2
    [pip3] pydata-google-auth==1.4.0
    [pip3] pydot==1.3.0
    [pip3] pydot-ng==2.0.0
    [pip3] pydotplus==2.0.2
    [pip3] PyDrive==1.3.1
    [pip3] pyemd==0.5.1
    [pip3] pyerfa==2.0.0.1
    [pip3] Pygments==2.6.1
    [pip3] pygobject==3.26.1
    [pip3] pylev==1.4.0
    [pip3] pymc==4.1.4
    [pip3] PyMeeus==0.5.11
    [pip3] pymongo==4.2.0
    [pip3] pymystem3==0.2.0
    [pip3] PyOpenGL==3.1.6
    [pip3] pyparsing==3.0.9
    [pip3] pyrsistent==0.18.1
    [pip3] pysimdjson==3.2.0
    [pip3] pysndfile==1.3.8
    [pip3] PySocks==1.7.1
    [pip3] pystan==3.3.0
    [pip3] pytest==3.6.4
    [pip3] python-apt==0.0.0
    [pip3] python-chess==0.23.11
    [pip3] python-dateutil==2.8.2
    [pip3] python-louvain==0.16
    [pip3] python-slugify==6.1.2
    [pip3] python-utils==3.3.3
    [pip3] pytz==2022.4
    [pip3] pyviz-comms==2.2.1
    [pip3] PyWavelets==1.3.0
    [pip3] PyYAML==6.0
    [pip3] pyzmq==23.2.1
    [pip3] qdldl==0.1.5.post2
    [pip3] qudida==0.0.4
    [pip3] regex==2022.6.2
    [pip3] requests==2.28.1
    [pip3] requests-oauthlib==1.3.1
    [pip3] resampy==0.4.2
    [pip3] rpy2==3.4.5
    [pip3] rsa==4.9
    [pip3] scikit-image==0.18.3
    [pip3] scikit-learn==1.0.2
    [pip3] scipy==1.7.3
    [pip3] screen-resolution-extra==0.0.0
    [pip3] scs==3.2.0
    [pip3] seaborn==0.11.2
    [pip3] Send2Trash==1.8.0
    [pip3] setuptools==57.4.0
    [pip3] setuptools-git==1.2
    [pip3] Shapely==1.8.4
    [pip3] six==1.15.0
    [pip3] sklearn-pandas==1.8.0
    [pip3] smart-open==5.2.1
    [pip3] snowballstemmer==2.2.0
    [pip3] sortedcontainers==2.4.0
    [pip3] soundfile==0.11.0
    [pip3] spacy==3.4.1
    [pip3] spacy-legacy==3.0.10
    [pip3] spacy-loggers==1.0.3
    [pip3] Sphinx==1.8.6
    [pip3] sphinxcontrib-serializinghtml==1.1.5
    [pip3] sphinxcontrib-websupport==1.2.4
    [pip3] SQLAlchemy==1.4.41
    [pip3] sqlparse==0.4.3
    [pip3] srsly==2.4.4
    [pip3] statsmodels==0.12.2
    [pip3] sympy==1.7.1
    [pip3] tables==3.7.0
    [pip3] tabulate==0.8.10
    [pip3] tblib==1.7.0
    [pip3] tenacity==8.1.0
    [pip3] tensorboard==2.8.0
    [pip3] tensorboard-data-server==0.6.1
    [pip3] tensorboard-plugin-wit==1.8.1
    [pip3] tensorflow==2.8.2+zzzcolab20220929150707
    [pip3] tensorflow-datasets==4.6.0
    [pip3] tensorflow-estimator==2.8.0
    [pip3] tensorflow-gcs-config==2.8.0
    [pip3] tensorflow-hub==0.12.0
    [pip3] tensorflow-io-gcs-filesystem==0.27.0
    [pip3] tensorflow-metadata==1.10.0
    [pip3] tensorflow-probability==0.16.0
    [pip3] termcolor==2.0.1
    [pip3] terminado==0.13.3
    [pip3] testpath==0.6.0
    [pip3] text-unidecode==1.3
    [pip3] textblob==0.15.3
    [pip3] thinc==8.1.2
    [pip3] threadpoolctl==3.1.0
    [pip3] tifffile==2021.11.2
    [pip3] toml==0.10.2
    [pip3] tomli==2.0.1
    [pip3] toolz==0.12.0
    [pip3] torch==1.12.1+cu113
    [pip3] torchaudio==0.12.1+cu113
    [pip3] torchsummary==1.5.1
    [pip3] torchtext==0.13.1
    [pip3] torchvision==0.13.1+cu113
    [pip3] tornado==5.1.1
    [pip3] tqdm==4.64.1
    [pip3] traitlets==5.1.1
    [pip3] tweepy==3.10.0
    [pip3] typeguard==2.7.1
    [pip3] typer==0.4.2
    [pip3] typing-extensions==4.1.1
    [pip3] tzlocal==1.5.1
    [pip3] ujson==5.5.0
    [pip3] uritemplate==3.0.1
    [pip3] urllib3==1.26.12
    [pip3] vega-datasets==0.9.0
    [pip3] wasabi==0.10.1
    [pip3] wcwidth==0.2.5
    [pip3] webargs==8.2.0
    [pip3] webencodings==0.5.1
    [pip3] websocket-client==1.4.1
    [pip3] Werkzeug==1.0.1
    [pip3] wheel==0.37.1
    [pip3] widgetsnbextension==3.6.1
    [pip3] wordcloud==1.8.2.2
    [pip3] wrapt==1.14.1
    [pip3] xarray==0.20.2
    [pip3] xarray-einstats==0.2.2
    [pip3] xgboost==0.90
    [pip3] xkit==0.0.0
    [pip3] xlrd==1.1.0
    [pip3] xlwt==1.3.0
    [pip3] yarl==1.8.1
    [pip3] yellowbrick==1.5
    [pip3] zict==2.2.0
    [pip3] zipp==3.8.1
    [conda] Could not collect
```

**Comment by ChrisCummins** [comment]
Hi @chunhualiao, thanks for the detailed bug report. The easiest workaround is to remove the `numpy` line from `examples/requirements.txt`. This will fix the dependency resolution error.

I'm working on a better fix in #767 but it's not ready to merge just yet.

Cheers,
Chris

**Comment by chunhualiao** [comment]
Your workaround works for me. Thanks.
